### DISCUSSION TITLE
WIX v4 how do we schedule util:CloseApplication immediately

### DISCUSSION THREAD

**Discussion by gloacai** [wixtoolset/issues#7845] [Questions]
We want to close all instances of an application for all users prior to the MSI updating the binary without having to prompt the admin to close the application or reboot the box. The eventual goal is to use it in automated deployments across multiple boxes simultaneously and there won't be a UI. The application in question has no user input and no persistent data that we need to worry about saving off. At first glance it looks like util:CloseApplication would do what we want but looking at the debug logs the underlying Custom Action is scheduled as "deferred" which restricts it to run after the prompt to the admin to close the running application.

To add more complexity to this, the MSI also delivers a service that launches instances of this application for all users, but we should be able to do a custom action to turn that off.

**Comment by EvgenPervenenko** [comment]
Hi, team, any thoughts about how to close application using util:CloseApplication before InstallValidate? Actual for Wix v5
